pkg/controller/deployment/deploment_helper_test.go

```diff
@@ -0,0 +1,398 @@
+				},
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			actualResources := mergeContainerResources(tc.sourceResources, tc.overrideResources)
+			require.Equal(t, tc.expected, actualResources)
+		})
+	}
+}
+
+func TestGetOverrideResourcesFor(t *testing.T) {
+	tests := []struct {
+		name                      string
+		certManagerObj            v1alpha1.CertManager
+		deploymentName            string
+		expectedOverrideResources v1alpha1.CertManagerResourceRequirements
+	}{
+		{
+			name: "get override resources of cert manager controller config",
+			certManagerObj: v1alpha1.CertManager{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cluster",
+				},
+				Spec: v1alpha1.CertManagerSpec{
+					ControllerConfig: &v1alpha1.DeploymentConfig{
+						OverrideResources: v1alpha1.CertManagerResourceRequirements{
+							Limits: corev1.ResourceList{
+								corev1.ResourceCPU: k8sresource.MustParse("10m"),
+							},
+							Requests: corev1.ResourceList{
+								corev1.ResourceMemory: k8sresource.MustParse("128Mi"),
+							},
+						},
+					},
+				},
+			},
+			deploymentName: certmanagerControllerDeployment,
+			expectedOverrideResources: v1alpha1.CertManagerResourceRequirements{
+				Limits: corev1.ResourceList{
+					corev1.ResourceCPU: k8sresource.MustParse("10m"),
+				},
+				Requests: corev1.ResourceList{
+					corev1.ResourceMemory: k8sresource.MustParse("128Mi"),
+				},
+			},
+		},
+		{
+			name: "get override resources of cert manager webhook config",
+			certManagerObj: v1alpha1.CertManager{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cluster",
+				},
+				Spec: v1alpha1.CertManagerSpec{
+					WebhookConfig: &v1alpha1.DeploymentConfig{
+						OverrideResources: v1alpha1.CertManagerResourceRequirements{
+							Limits: corev1.ResourceList{
+								corev1.ResourceCPU: k8sresource.MustParse("10m"),
+							},
+							Requests: corev1.ResourceList{
+								corev1.ResourceMemory: k8sresource.MustParse("128Mi"),
+							},
+						},
+					},
+				},
+			},
+			deploymentName: certmanagerWebhookDeployment,
+			expectedOverrideResources: v1alpha1.CertManagerResourceRequirements{
+				Limits: corev1.ResourceList{
+					corev1.ResourceCPU: k8sresource.MustParse("10m"),
+				},
+				Requests: corev1.ResourceList{
+					corev1.ResourceMemory: k8sresource.MustParse("128Mi"),
+				},
+			},
+		},
+		{
+			name: "get override resources of cert manager cainjector config",
+			certManagerObj: v1alpha1.CertManager{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cluster",
+				},
+				Spec: v1alpha1.CertManagerSpec{
+					CAInjectorConfig: &v1alpha1.DeploymentConfig{
+						OverrideResources: v1alpha1.CertManagerResourceRequirements{
+							Limits: corev1.ResourceList{
+								corev1.ResourceCPU: k8sresource.MustParse("10m"),
+							},
+							Requests: corev1.ResourceList{
+								corev1.ResourceMemory: k8sresource.MustParse("128Mi"),
+							},
+						},
+					},
+				},
+			},
```

Comment:
Can a add another or a couple test case(s) to test the behaviour we discussed in case of:  https://github.com/openshift/cert-manager-operator/pull/136#discussion_r1281566526

Reply:
Those are added here: https://github.com/openshift/cert-manager-operator/pull/136/files#diff-fe3b58419cf80f35e0dc128fc74a9aa700807ea529635eafa1e44c8d1f39ebd6R23